benches/store.rs

```diff
@@ -0,0 +1,207 @@
+fn insert(c: &mut Criterion) {
+    let mut group = c.benchmark_group("insert");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        // MerkleTree constructor is optimized to work with vectors. Create a new copy of the data
+        // nd pass it to the benchmark function
+        group.bench_function(
+            BenchmarkId::new(format!("MerkleTree/new/{size}"), size),
+            |b| {
+                b.iter_batched(
+                    || leaves.iter().map(|v| v.into()).collect::<Vec<Word>>(),
+                    |l| black_box(MerkleTree::new(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+
+        // This could be done with `bench_with_input`, however to remove variables while comparing
+        // with MerkleTree it is using `iter_batched`
+        group.bench_function(
+            BenchmarkId::new(format!("MerkleStore/with_merkle_tree/{size}"), size),
+            |b| {
+                b.iter_batched(
+                    || leaves.iter().map(|v| v.into()).collect::<Vec<Word>>(),
+                    |l| black_box(MerkleStore::new().with_merkle_tree(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+
+        group.bench_function(
+            BenchmarkId::new(format!("SimpleSmt/new/{size}"), size),
+            |b| {
+                b.iter_batched(
+                    || {
+                        leaves
+                            .iter()
+                            .enumerate()
+                            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+                            .collect::<Vec<(u64, Word)>>()
+                    },
+                    |l| black_box(SimpleSmt::new(63).unwrap().with_leaves(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+
+        group.bench_function(
+            BenchmarkId::new(format!("MerkleStore/with_sparse_merkle_tree/{size}"), size),
+            |b| {
+                b.iter_batched(
+                    || {
+                        leaves
+                            .iter()
+                            .enumerate()
+                            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+                            .collect::<Vec<(u64, Word)>>()
+                    },
+                    |l| black_box(MerkleStore::new().with_sparse_merkle_tree(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+    }
+}
+
+/// Benchmarks getting a node on each of the different storage backends
+///
+/// Since MerkleTree can only be created when a power-of-two number of elements is used, the sample
+/// sizes are limited to that
+fn get_node(c: &mut Criterion) {
```

Comment:
I would probably break this into two separate comparisons:

1. `MerkleTree` vs. `MerkleStore` instantiated with `with_merkle_tree`.
2. `SimpleSmt` of depth 63 vs. `MerkleStore` instantiated with `with_sparse_merkle_tree` (I believe this will assume depth 63 as well, but I wonder if we should make the depth be an explicit parameter).

Reply:
split benchmarks